api/envoy/config/filter/http/jwt_authn/v2alpha/config.proto

```diff
@@ -183,44 +183,82 @@ message JwtHeader {
   string value_prefix = 2;
 }
 
+// This message specifies a Jwt requirement.
+//
+// .. code-block:: yaml
+//
+//    required_issuers: [jwt_issuer1, jwt_issuer2]
+//
+// PerRouteConfig for jwt_authn filter uses this message.
+message JwtRequirement {
+  // Specify list of required Jwt issuers. If list is empty,  Jwt is NOT required.
+  // If any of required issuers is missing, or its verification fails, the request will
+  // be rejected.
+  // If a Jwt issuer is not required and is in the config, it will be verified.
+  // But it will not fail the request if its verification fails.
+  repeated string required_issuers = 1;
+}
+
+// This message specifies a Jwt requirement for a specific Route condition.
+// For example:
+//
+// .. code-block:: yaml
+//
+//    - match:
+//         prefix: "/healthz"
+//      requirement:
+//         required_issuers: []
+//    - match:
+//         prefix: "/"
+//      requirement:
+//         required_issuers: [jwt_issuer1]
+//         allow_missing_or_failed: true
+//
+message RouteRequirement {
+  // The route matching parameter.  Only match is satisfied, the other fields will apply.
+  //
+  // For example: following match will match all requests.
+  //
+  // .. code-block:: yaml
+  //
+  //    match:
+  //       prefix: "/"
+  //
+  envoy.api.v2.route.RouteMatch match = 1 [(validate.rules).message.required = true];
+
+  // Specifies the Jwt requiement for matched routes.
+  JwtRequirement requirement = 2;
+}
+
 // This is the Envoy HTTP filter config for JWT authentication.
 // [#not-implemented-hide:]
 message JwtAuthentication {
   // List of JWT rules to valide.
   repeated JwtRule rules = 1;
```

Comment:
I would maybe rename this, since they are no longer rules but provider defintions.

Reply:
Actually I don't know anymore, maybe rules is okay, you guys can discuss internally what the naming should be.

Reply:
This config is used by Istio now.   Field change requires code change.

Reply:
Changed to Provider